homeassistant/components/envisalink.py

```diff
@@ -0,0 +1,208 @@
+                                          _pass,
+                                          _zone_dump,
+                                          _keep_alive)
+
+    def login_fail_callback(data):
+        """Callback for when the evl rejects our login."""
+        _LOGGER.error("The envisalink rejected your credentials.")
+        _connect_status['fail'] = 1
+
+    def connection_fail_callback(data):
+        """Network failure callback."""
+        _LOGGER.error("Could not establish a connection with the envisalink.")
+        _connect_status['fail'] = 1
+
+    def connection_success_callback(data):
+        """Callback for a successful connection."""
+        _LOGGER.info("Established a connection with the envisalink.")
+        _connect_status['success'] = 1
+
+    def zones_updated_callback(data):
+        """Handle zone timer updates."""
+        _LOGGER.info("Envisalink sent a zone update event.  Updating zones...")
+        dispatcher.send(signal=SIGNAL_ZONE_UPDATE,
+                        sender=None,
+                        zone=data)
+
+    def alarm_data_updated_callback(data):
+        """Handle non-alarm based info updates."""
+        _LOGGER.info("Envisalink sent new alarm info. Updating alarms...")
+        dispatcher.send(signal=SIGNAL_KEYPAD_UPDATE,
+                        sender=None,
+                        partition=data)
+
+    def partition_updated_callback(data):
+        """Handle partition changes thrown by evl (including alarms)."""
+        _LOGGER.info("The envisalink sent a partition update event.")
+        dispatcher.send(signal=SIGNAL_PARTITION_UPDATE,
+                        sender=None,
+                        partition=data)
+
+    def stop_envisalink(event):
+        """Shutdown envisalink connection and thread on exit."""
+        _LOGGER.info("Shutting down envisalink.")
+        EVL_CONTROLLER.stop()
+
+    def start_envisalink(event):
+        """Startup process for the envisalink."""
+        EVL_CONTROLLER.start()
+        for _ in range(10):
+            if 'success' in _connect_status:
+                hass.bus.listen_once(EVENT_HOMEASSISTANT_STOP, stop_envisalink)
+                return True
+            elif 'fail' in _connect_status:
+                return False
+            else:
+                time.sleep(1)
+
+        _LOGGER.error("Timeout occurred while establishing evl connection.")
+        return False
+
+    EVL_CONTROLLER.callback_zone_timer_dump = zones_updated_callback
+    EVL_CONTROLLER.callback_zone_state_change = zones_updated_callback
+    EVL_CONTROLLER.callback_partition_state_change = partition_updated_callback
+    EVL_CONTROLLER.callback_keypad_update = alarm_data_updated_callback
+    EVL_CONTROLLER.callback_login_failure = login_fail_callback
+    EVL_CONTROLLER.callback_login_timeout = connection_fail_callback
+    EVL_CONTROLLER.callback_login_success = connection_success_callback
+
+    _result = start_envisalink(None)
+    if _result:
```

Comment:
_This is more a tip than something that needs fixing._

You can use a guard clause here because no code will be executed if result is not truthy, it will will save you an indentation.

```python
_result = …

if not _result:
    return _result  # Which I guess could be replaced with None?

if _partitions:
    …
```